File: module-services/src/main/java/eu/geekhome/services/configurable/MaxStringLengthValidator.java
package eu.geekhome.services.configurable;


import eu.geekhome.services.localization.Resource;

public class MaxStringLengthValidator implements Validator<String> {

    private final int _maxLength;

    public MaxStringLengthValidator(int maxLength) {
        _maxLength = maxLength;
    }

    @Override
    public Resource getReason() {
        return new Resource("Max length is " + _maxLength + " characters",
                "Maksymalna długość to " + _maxLength + " znaków");
    }

    @Override
    public boolean validate(String fieldValue) {
        if (fieldValue == null) {
            return true;
        }

        return fieldValue.length() <= _maxLength;
    }
}

File: module-services/src/main/java/eu/geekhome/services/configurable/MaxStringLengthValidator.kt
package eu.geekhome.services.configurable

import eu.geekhome.services.localization.Resource

class MaxStringLengthValidator(val maxLength: Int) : Validator<String?> {
    override val reason: Resource
        get() = Resource(
            "Max length is $maxLength characters",
            "Maksymalna długość to $maxLength znaków"
        )

    override fun validate(fieldValue: String?): Boolean {
        return if (fieldValue == null) {
            true
        } else fieldValue.length <= maxLength
    }
}